### PR TITLE
Finish BrowseService and rename method to better names

### PR DESCRIPTION
**Pull request by topnessman**
Rename get/find to retrieve that better illustrates this process is
fetching existing staff from database

### CODE REVIEW

**Review by dgcaccam — APPROVED** [review]
looks good

File: src/main/java/edu/uwaterloo/ece658/service/BrowseService.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+     * user
+     */
+    public Map<Tag, List<Photo>> fetchPhotosClassifiedBySubscribedTagsOfUser(String userName) {
+        User user = userFacade.retrieveUserByUserName(userName);
+        Map<Tag, List<Photo>> result = new HashMap<>();
+        for (Tag tag : user.getSubscribedTags()) {
+            // User explicitly subscribed to some tags => Directly show photos
+            // under these tags is ok. This includes UserTag that the user
+            // has subscribed to.
+            result.put(tag, tag.getPhotosUnderThisTag());
```
Comment (dgcaccam) [inline src/main/java/edu/uwaterloo/ece658/service/BrowseService.java]: As an afterthought.... Where do we deal with private photos? As a private photo still may be under one of these tags. Will the front end not show these? Or should they be filtered out before passing the Map

Comment (topnessman) [reply src/main/java/edu/uwaterloo/ece658/service/BrowseService.java]: Good question.
The point of private photo is only to say that "please don't notify my private photo to any one that subscribed to the tags of my private photo". notifyXXX() methods can expose private methods so we need to have a check there. Here this method can still leak the private photos. So I need to have a check, too.
Doing it at the backend should be correct. Even though you front-end doesn't show private photos, client can still find a way to get private photos if only your backend returns them.

**Review by dgcaccam — APPROVED** [review]
cool

**Pull request merged**